### DISCUSSION TITLE
Do we really need useSyncExternalStore?

### DISCUSSION THREAD

**Discussion by Jokcy** [reduxjs/react-redux#1860] [General]
As I know for redux situation, normal setState is enough. For React 18 concurrent rendering will only be triggered by a few APIs like

- startTransition
- useDeferreredValue

So in connect of react-redux, we will not use these kind of APIs, and just `setState`, which will fire sync updates. So why we need to use `useSyncExternalStore`?

**Comment by markerikson** [comment]
Because we need to avoid tearing in any concurrent rendering situation.  You might want to read these prior resources:

- https://github.com/reactjs/rfcs/blob/main/text/0147-use-mutable-source.md
- https://github.com/reactwg/react-18/discussions/86

**Reply by Jokcy** [reply]
Is there any example that without `useSyncExternalStore` react-redux will render state inconsistent?
